test/hotspot/jtreg/compiler/intrinsics/base64/TestBase64.java

```diff
@@ -59,15 +61,40 @@ public static void main(String[] args) throws Exception {
         if (!Compiler.isIntrinsicAvailable(CompilerWhiteBoxTest.COMP_LEVEL_FULL_OPTIMIZATION, "java.util.Base64$Encoder", "encodeBlock", byte[].class, int.class, int.class, byte[].class, int.class, boolean.class)) {
             throw new SkippedException("Base64 intrinsic is not available");
         }
-        int iters = (args.length > 0 ? Integer.valueOf(args[0]) : 100000);
+        int iters = (args.length > 0 ? Integer.valueOf(args[0]) : 5_000);
         System.out.println(iters + " iterations");
 
-        test0(Base64Type.BASIC, Base64.getEncoder(), Base64.getDecoder(),"plain.txt", "baseEncode.txt", iters);
-        test0(Base64Type.URLSAFE, Base64.getUrlEncoder(), Base64.getUrlDecoder(),"plain.txt", "urlEncode.txt", iters);
-        test0(Base64Type.MIME, Base64.getMimeEncoder(), Base64.getMimeDecoder(),"plain.txt", "mimeEncode.txt", iters);
+        initNonBase64Arrays();
+
+        warmup();
+
+        test0(FileType.ASCII, Base64Type.BASIC, Base64.getEncoder(), Base64.getDecoder(),"plain.txt", "baseEncode.txt", iters);
+        test0(FileType.ASCII, Base64Type.URLSAFE, Base64.getUrlEncoder(), Base64.getUrlDecoder(),"plain.txt", "urlEncode.txt", iters);
+        test0(FileType.ASCII, Base64Type.MIME, Base64.getMimeEncoder(), Base64.getMimeDecoder(),"plain.txt", "mimeEncode.txt", iters);
+
+        test0(FileType.HEXASCII, Base64Type.BASIC, Base64.getEncoder(), Base64.getDecoder(),"longLineHEX.txt", "longLineBaseEncode.txt", iters);
+        test0(FileType.HEXASCII, Base64Type.URLSAFE, Base64.getUrlEncoder(), Base64.getUrlDecoder(),"longLineHEX.txt", "longLineUrlEncode.txt", iters);
+        test0(FileType.HEXASCII, Base64Type.MIME, Base64.getMimeEncoder(), Base64.getMimeDecoder(),"longLineHEX.txt", "longLineMimeEncode.txt", iters);
+    }
+
+    private static void warmup() {
+        final int warmupCount = 20_000;
+        final int bufSize = 60;
+        byte[] srcBuf = new byte[bufSize];
+        byte[] encBuf = new byte[(bufSize / 3) * 4];
+        byte[] decBuf = new byte[bufSize];
+
+        ran.nextBytes(srcBuf);
+
+        // This should be enough to get both encodeBlock() and
+        // decodeBlock() compiled on the highest tier.
```

Comment:
It's actually encode() and decode() which should get compiled. You should see them when testing with -XX:+PrintCompilation. And you should see usage of the intrinsics by -XX:+PrintInlining.

Reply:
Ah, useful tips!  Thanks!

I will make this comment change.